pkg/apis/operator/base/common.go

```diff
@@ -137,15 +137,20 @@ type CommonSpec struct {
 	// +optional
 	DeprecatedResources []ResourceRequirementsOverride `json:"resources,omitempty"`
 
+	// DEPRECATED. Use components
 	// DeploymentOverride overrides Deployment configurations such as resources and replicas.
 	// +optional
-	DeploymentOverride []DeploymentOverride `json:"deployments,omitempty"`
+	DeploymentOverride []WorkloadOverride `json:"deployments,omitempty"`
+
+	// ComponentsOverride overrides workloads configurations such as resources and replicas.
+	// +optional
+	ComponentsOverride []WorkloadOverride `json:"workloads,omitempty"`
```

Comment:
I guess we can have WorkloadsOverride in the future?

Reply:
I will rename it now to `Workloads`